esi_ui/static/dashboard/project/esi/nodes/esi-nodes-table.controller.js

```diff
@@ -204,21 +205,48 @@
       var launchContext = {
         node: node
       };
-
+    
       manageNetworksModalService.open(launchContext)
-      .then(function (response) {
-        if (response.action === 'attach') {
-          nodesService.networkAttach(node, response.attach)
-          .then(function() {
-            init();
-          });
-        } else {
-          nodesService.networkDetach(node, response.detach)
-          .then(function() {
-            init();
-          });
-        }
-      });
+        .then(function (response) {
+          if (response.action === 'attach') {
+            node.network_operation = 'attaching';
+            toastService.add('success', 'Attaching network, please wait...');
+            
+            nodesService.networkAttach(node, response.attach)
+              .then(function() {
+                $timeout(function() {
+                  node.network_operation = null;
```

Comment:
Currently, the success toast comes and then it takes some time for it to load up the nodes/networks using init() and reflect the updated state. To fix this, this line (218) along with the toast message line (219) should be written after the init() function is called (on line 220) as the nodes/networks get updated using init(), so the "loaded" state should become false after that.

Reply:
okay. I will make it that way for both of the methods.